miio/viomivacuum.py

```diff
@@ -0,0 +1,175 @@
+import logging
+from collections import defaultdict
+from datetime import timedelta
+from enum import Enum
+
+import click
+
+from .click_common import command, format_output
+from .device import Device
+from .utils import pretty_seconds
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class ViomiVacuumSpeed(Enum):
+    Silent = 0
+    Standard = 1
+    Medium = 2
+    Turbo = 3
+
+
+class ViomiVacuumState(Enum):
```

Comment:
It seems to sometimes also return 0 for state (idle, not docked), subquent queries return ViomiVacuumState.Idle (1).

```
DEBUG:miio.device:192.168.178.82:54321 >>: {'id': 405, 'method': 'get_prop', 'params': ['run_state', 'mode', 'err_state', 'battary_life', 'box_type', 'mop_type', 's_time', 's_area', 'suction_grade', 'water_grade', 'remember_map', 'has_map', 'is_mop', 'has_newmap']}
DEBUG:miio.device:192.168.178.82:54321 (ts: 2019-11-23 09:13:34, id: 405) << {'result': [0, 0, 0, 76, 0, 0, 31, 18.13, 0, 12, 1, 1, 1, 0], 'id': 405}
ValueError: 0 is not a valid ViomiVacuumState
```

Reply:
Fixed.